src/runtime/manifest-parser.peg

```diff
@@ -247,37 +273,13 @@ Interface "an interface"
     });
   }
 
-InterfaceItem
-  = InterfaceSlot
-  / InterfaceArgument
-
 InterfaceArgument
-  = name:NameWithColon direction:Direction? type:(whiteSpace? ParticleHandleConnectionType)? eolWhiteSpace
-  { return buildInterfaceArgument(name, direction || 'any', optional(type, t => t[1], null)); }
-  / direction:Direction type:(whiteSpace? ParticleHandleConnectionType)? eolWhiteSpace
-  { return buildInterfaceArgument('*', direction || 'any', optional(type, t => t[1], null)); }
-  / type:(whiteSpace? ParticleHandleConnectionType) eolWhiteSpace
-  { return buildInterfaceArgument('*', 'any', type); }
-
-InterfaceSlot
-  = name:NameWithColon? direction:('consumes' / 'provides') isOptional:'?'? type:(whiteSpace SlandleType)? eolWhiteSpace
-  {
-    let isSet = false;
-    if (type) {
-      type = type[1]; // remove the whitespace.
-      isSet = type.isSet;
-      type.fields.forEach(({name, value}) => {
-        error(`interface slots do not currently support fields`);
-      });
-    }
-    return toAstNode<AstNode.InterfaceSlot>({
-      kind: 'interface-slot',
-      name,
-      isRequired: !isOptional,
-      direction,
-      isSet,
-    });
-  }
+  = name:NameWithColon direction:(Direction / SlotDirection)? isOptional:'?'? type:(whiteSpace? ParticleHandleConnectionType)? eolWhiteSpace
```

Comment:
It might be useful to add a comment above this rule, along the lines of "This rule is effectively capturing `name? direction? '?'? type?` but is split into multiple capture clauses because the combined one is able to match against an empty string."

Reply:
I found an alternative resolution which I think is simpler.
The underlying problem, that a handle connection can be 'empty' was made worse by the indentation being possibly empty (which gave us the infinite loop issue).

Making indents incapable of parsing empty string and making potentially empty indentation into optional indentations, makes the parser aware that this parser node cannot infinite loop.
Then we can use a guard to ensure that empty lines with appropriate indentation doesn't accidentally create handle connections.